Extend/Controller/Admin/PaymentMain.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Extend\Controller\Admin;
+
+use OxidEsales\Eshop\Core\Registry;
+use Wirecard\Oxid\Extend\Model\Payment;
+use Wirecard\Oxid\Model\SofortPaymentMethod;
+use Wirecard\PaymentSdk\Config\Config;
+use Wirecard\PaymentSdk\TransactionService;
+
+/**
+ * Controls the payment method config view.
+ *
+ * @mixin \OxidEsales\Eshop\Application\Controller\Admin\PaymentMain
+ */
+class PaymentMain extends PaymentMain_parent
+{
+    /**
+     * @inheritdoc
+     *
+     * @return string
+     */
+    public function render()
+    {
+        $sParentReturn = parent::render();
+
+        if ($this->_isCustomPaymentMethod()) {
+            $sFcn = Registry::getRequest()->getRequestParameter('fnc');
+            $this->_aViewData["bConfigNotValid"] = ($sFcn === 'save' || $sFcn === 'addfield')
+                && !$this->_isSavePossible();
+        }
+
+        return $sParentReturn;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return void
+     */
+    public function save()
+    {
+        if ($this->_isCustomPaymentMethod() && !$this->_isSavePossible()) {
+            $sOxId = $this->getEditObjectId();
+            $this->setEditObjectId($sOxId);
+            return;
+        }
+
+        parent::save();
+    }
+
+    /**
+     * Checks if it is possible to save the config
+     *
+     * @return bool
+     */
+    private function _isSavePossible()
+    {
+        $aParams = Registry::getRequest()->getRequestParameter('editval');
+
+        $bCredentialsValid = $this->_isFormCredentialsValid($aParams);
+
+        $sCountryCode = $aParams['oxpayments__wdoxidee_countrycode'];
+        $bCountryCodeValid = $aParams['oxpayments__oxid'] !== SofortPaymentMethod::getName(true)
+            || (preg_match('/^[a-z]{2}_[a-z]{2}$/', $sCountryCode) === 1);
```

Comment:
This condition is Sofort-specific. I would create a validation helper and put this in there. But keep it for now, we'll generalize when another method requires validation.